Sources/MapboxCoreNavigation/RouteLegProgress.swift

```diff
@@ -142,11 +142,11 @@ open class RouteLegProgress: Codable {
      */
     public init(leg: RouteLeg, stepIndex: Int = 0, spokenInstructionIndex: Int = 0) {
         self.leg = leg
-        self.stepIndex = stepIndex
         
         precondition(leg.steps.indices.contains(stepIndex), "It's not possible to create RouteLegProgress without any steps or when stepIndex is higher than steps count.")
         
         currentStepProgress = RouteStepProgress(step: leg.steps[stepIndex], spokenInstructionIndex: spokenInstructionIndex)
+        self.stepIndex = stepIndex
```

Comment:
Not sure I understand how the previous location of this statement was causing the crash in #3062 or the nil in #2997 (which itself is strange because the property is non-optional). Within this initializer, the `stepIndex` argument shadows the `stepIndex` instance property, and nothing else in this initializer reads from the `RouteLegProgress.stepIndex` property. (Neither `RouteLegProgress.leg` nor `RouteLegProgress.currentStepProgress` has a setter that depends on `RouteLegProgress.stepIndex`.) Since nothing reads from the property, it should’t matter where we set it, as long as it’s after setting `RouteLegProgress.leg`.

Reply:
for the #2997, the break issue comes in the following and several other places with the `RouteLegProgress.init(leg:stepIndex:spokenInstructionIndex:)`, and the `nil` of the `routeProgress.currentLegProgress.currentStepProgress`. 
https://github.com/mapbox/mapbox-navigation-ios/blob/c32cf61f0038dceb05984e56a7913821dda23c29/Sources/MapboxNavigation/NavigationViewController.swift#L1054-L1057
where after setting the initializing of the `RouteLegProgress` and reach the `RouteLegProgress.currentStep`, then the `currentStep` is nil. I remembered that I've checked the index of the `legIndex` and `stepIndex`,and it happened just after the navigation session start, they're not out of the bound.  But after that fix, and the problem could not reproduce anymore.